### ISSUE TITLE
Suggestion: Contextually relevant messages

### ISSUE THREAD

**Issue by Sirs0ri** [ericaxu/gemmy#2]
This idea came from Gemmy's "You have plugins that you can update!" message, which coincidentally showed up just after I had updated my plugins, ie. at a time where I definitely didn't have any updates available.

It would be cool if there was an option to filter/populate the list of possible messages dynamically before selecting a message. Then it would for example be possible to start an update-check (`app.plugins.checkForUpdates()`), and only show the updates related message if there's actually any new updates.

This feature could also be extended to make Gemmy actually useful in certain conditions, some other (more or less helpful) contextually relevant suggestions could be: 
- if the user's in an empty note, suggest adding some content, 
- a comment about how many/few tabs the user has open, 
- comments about the dark/light theme ("Come to the dark side" for light theme users comes to mind), etc

Gemmy is probably my favorite April fools joke this year, I'd hope to see it grow over time! :)

**Comment by ericaxu** [comment]
Thank you for your feedback, will definitely consider it!

**Comment by Sergei-Korneev** [comment]
Hello, guys. BTW,  is there any method to check for plugins/app updates which returns something like true/false/array? The app.plugins.checkForUpdates() only shows a notification and returns undefined.

**Comment by Sirs0ri** [comment]
Yes, there kinda is. `app.plugins.updates` is either an empty object, or it has info about outstanding updates. But I don't think the API exposes a method that checks for updates and returns the result right away. You'd have to call `app.plugins.checkForUpdates()`, await the result and then check `app.plugins.updates`. 

I currently have outstanding updates and it looks like this:

```json
{
  "obsidian-leaflet-plugin": {
    "repo": "valentine195/obsidian-leaflet-plugin",
    "version": "5.5.0",
    "manifest": {
      "id": "obsidian-leaflet-plugin",
      "name": "Obsidian Leaflet",
      "description": "Interactive maps inside your notes",
      "version": "5.5.0",
      "minAppVersion": "0.12.12",
      "author": "Jeremy Valentine",
      "repo": "valentine195/obsidian-leaflet-plugin",
      "isDesktopOnly": false
    }
  }
}
```
